js/apps/admin-ui/src/realm-settings/RealmSettingsTabs.tsx

```diff
@@ -361,13 +364,15 @@ export const RealmSettingsTabs = () => {
           >
             <KeysTab />
           </Tab>
-          <Tab
-            title={<TabTitleText>{t("events")}</TabTitleText>}
-            data-testid="rs-realm-events-tab"
-            {...eventsTab}
-          >
-            <EventsTab realm={realm!} />
-          </Tab>
+          {canViewOrManageEvents ? (
+            <Tab
+              title={<TabTitleText>{t("events")}</TabTitleText>}
+              data-testid="rs-realm-events-tab"
+              {...eventsTab}
+            >
+              <EventsTab realm={realm!} />
+            </Tab>
+          ) : null}
```

Comment:
This should be a cleaner way to do it.  Give this a try:
```suggestion
          {canViewOrManageEvents && (
            <Tab
              title={<TabTitleText>{t("events")}</TabTitleText>}
              data-testid="rs-realm-events-tab"
              {...eventsTab}
            >
              <EventsTab realm={realm!} />
            </Tab>
          )}
```